sst/SimEngCoreWrapper.cc

```diff
@@ -223,16 +235,53 @@ std::vector<std::string> SimEngCoreWrapper::splitArgs(std::string strArgs) {
 
 void SimEngCoreWrapper::fabricateSimEngCore() {
   output_.verbose(CALL_INFO, 1, 0, "Setting up SimEng Core\n");
-
-  // Create the instance of the core to be simulated
   if (simengConfigPath_ != "") {
+#ifdef SIMENG_ENABLE_SST_TESTS
+    if (assembleWithSource_) {
+      output_.verbose(CALL_INFO, 1, 0,
+                      "Assembling source instructions using LLVM\n");
+      Assembler assemble = Assembler(source_);
+      coreInstance_ = std::make_unique<simeng::CoreInstance>(
+          assemble.getAssembledSource(), assemble.getAssembledSourceSize(),
+          simengConfigPath_);
+    } else {
+      coreInstance_ = std::make_unique<simeng::CoreInstance>(
+          simengConfigPath_, executablePath_, executableArgs_);
+    }
+#else
     coreInstance_ = std::make_unique<simeng::CoreInstance>(
         simengConfigPath_, executablePath_, executableArgs_);
+#endif
   } else {
+#ifdef SIMENG_ENABLE_SST_TESTS
+    std::string a64fxConfigPath = std::string(SIMENG_BUILD_DIR) +
+                                  "/simeng-configs/sst-cores/a64fx-sst.yaml";
+    output_.verbose(
+        CALL_INFO, 1, 0,
+        "No config path provided so defaulting to a64fx-sst.yaml\n");
+    if (assembleWithSource_) {
+      output_.verbose(CALL_INFO, 1, 0,
+                      "Assembling source instructions using LLVM\n");
+      Assembler assemble = Assembler(source_);
+      coreInstance_ = std::make_unique<simeng::CoreInstance>(
+          assemble.getAssembledSource(), assemble.getAssembledSourceSize(),
+          a64fxConfigPath);
+    } else {
+      coreInstance_ = std::make_unique<simeng::CoreInstance>(
+          a64fxConfigPath, executablePath_, executableArgs_);
+    }
+#else
     coreInstance_ = std::make_unique<simeng::CoreInstance>(executablePath_,
                                                            executableArgs_);
+#endif
+  }
+  if (coreInstance_->getSimulationMode() !=
+      simeng::SimulationMode::OutOfOrder) {
+    std::cerr << "[SimEng] SimEng only supports out-of-order cores with SST "
```

Comment:
To be consistent with the docs, maybe we should refer to the OoO sim mode as an archetype not a core https://uob-hpc.github.io/SimEng/developer/models/index.html#archetypes

Reply:
Done. Please check